### DISCUSSION TITLE
Question  about Groovy console

### DISCUSSION THREAD

**Discussion by ebocher** [dbeaver/dbeaver#16662] [Ideas]
Dear dbeaver community,

Is there any plan to add a Groovy console as the SQL-Editor panel?
Thank you and bravo for this excellent tool

**Comment by ebocher** [comment]
For information, 
We have developed  a basic Groovy console.
See https://github.com/orbisgis/orbisrcp/tree/master/plugins/org.orbisgis.groovyeditor

**Comment by sken130** [comment]
I second this request.

**Comment by sken130** [comment]
Further suggestions:

Can the Groovy console manipulate data in the SQL results window?
Can the Groovy console add custom entries in the right click menus of the SQL results window?

**Comment by ebocher** [comment]
> Can the Groovy console manipulate data in the SQL results window?

Must be feasible  using dbeaver services to access a jdbc connection. Not tested yet ;-)

> Can the Groovy console add custom entries in the right click menus of the SQL results window?

Same idea than  previously. @serge-rider, can you give us an example of how to access dbeaver services?  or @spalominos

**Comment by ebocher** [comment]
Dear DBeaver friends,

On our side, we use dbeaver and the groovy console extensively to automate tasks like rendering graphics with vega-lite.
The DBeaver groovy plugin is here https://github.com/orbisgis/orbisrcp
We 'd like to acces to Eclipse and DBeaver classes in order, for example, to show html vega-lite result in a window as dbeaver does with the map.
So the question is. Is there a way a get access to Eclipse-DBeaver OSGI services ?
I think of something like this : 
```java
import org.jkiss.dbeaver.registry.RegistryUtil

def registry = RegistryUtil.getRegistry()
Browser bundleContext = (Browser) registry.bundleContext
```
Best regards
